### ISSUE TITLE
Modifying headers

### ISSUE THREAD

**Issue by kevinkhill** [moul/node-slumber#4]
I didn't see noted anywhere how to modify the headers on each api request. I also tried looking through the source, but coffee -> js is a bit cryptic to me.

I want to be able to add a object such as:
```
headers: {
    'PRIVATE-TOKEN': 'not-a-real-private-token'
}
```
that will be added to each API request.

I tried adding it after the base API url:
```
var gitlabApi = slumber.API('http://my.gitlab.instance.com/api/v3/', {
    request_opts: {
      headers: {
        'PRIVATE-TOKEN': 'not-a-real-private-token'
      }
    }
  });
```
and I can see them added to the array when I ```console.log(gitlabApi.opts);```
```
{ request_opts: { headers: { 'PRIVATE-TOKEN': 'not-a-real-private-token' } },
  base_url: 'http://my.gitlab.instance.com/api/v3/',
  append_slash: true,
  auth: null,
  format: 'json',
  serializer:
   { default: 'json',
     get_content_type: [Function],
     dumps: [Function],
     loads: [Function],
     get_serializer: [Function],
     serializers: { json: [Object] } } }
```
but when the request is fired, I am still unauthorized and it doesn't seem like the headers are getting attached.

When I use the plain request library, and pass the headers as such:
```
request({
    uri: 'http://my.gitlab.instance.com/api/v3/users/keys',
    method: 'get',
    headers: {
        'PRIVATE-TOKEN': 'not-a-real-private-token'
    }
})
```
it works, so I know that my key is good and the gitlab api is working.

Any insight would be helpful, thank you!

**Comment by moul** [comment]
Hi @kevinkhill,

I refactored a little bit the code so you can pass headers more easily, you can pass headers when creating the API object or when calling method.

For an example, checkout the tests here: https://github.com/moul/node-slumber/blob/master/tests/test.js#L423-L485

**Comment by kevinkhill** [comment]
Fantastic :) works like a charm. I created my api as such:
```
var gitlabApi = slumber.API('http://my.gitlab.instance.com/api/v3/', {
      headers: {
          'PRIVATE-TOKEN': 'not-a-real-private-token'
      }
  });
```
and it worked just as expected.

Thank you for the quick fix!

**Comment by kevinkhill** [comment]
So it worked great for my gitlab api, but when trying to use it for github, I found another issue that was kind of difficult to debug.

By adding my own headers, it over-writes the User-Agent, unless your library doesn't have a default user-agent?

My second question was, could I get some more documentation for the callback params for a .get() request? or any request in fact?

I tried ```function(data)``` first, and:
```
data  = null
```

then I tried ```function(err, data)``` and it was:
```
err = { statusCode: 403 }
data = null
```

then I tried ```function(err, data, extra)``` and it was:
```
err = { statusCode: 403 }
data = {it looks like tons of properties that I was having trouble sifting through}
extra = null
```
within those properties, I found ```data.body``` which contained the error I was having
```
'Request forbidden by administrative rules. Please make sure your request has a User-Agent header (http://developer.github.com/v3/#user-agent-required). Check https://developer.github.com for other possible causes.'
```

**Comment by moul** [comment]
Yes you're totally right, a mini explanation before I update the `README.md` file

I use the `debug` npm mobule, so you can have debug by setting up the environment variable `DEBUG='*'`.

About the returned values, it depends on the arity of your callback : 
- with an arity of 1: `function(err)`
- with an arity of 2: `function(err, processedData)`
- with an arity of 3: `function(err, fullRequestResponse, processedData)`

It's clearly a little bit strange and needs at least a clean explanation :)
Btw, if you want an example of usage of this module, you can give a look to https://github.com/node-gitlab/node-gitlab

I will also add a default `User-Agent` header soon, thanks

**Comment by kevinkhill** [comment]
Fantastic, thank you for the explanation too

**Comment by moul** [comment]
I just released a 0.6.0 version on npm with this fix and other ones from #3
